tensorflow_addons/optimizers/lookahead.py

```diff
@@ -108,11 +108,13 @@ def _init_op(self, var):
     def _look_ahead_op(self, var):
         var_dtype = var.dtype.base_dtype
         slow_var = self.get_slot(var, 'slow')
-        local_step = tf.cast(self.iterations + 1, var_dtype)
-        sync_period = self._get_hyper('sync_period', local_step.dtype)
+        local_step = tf.cast(self.iterations + 1, tf.dtypes.int64)
+        sync_period = self._get_hyper('sync_period', tf.dtypes.int64)
         slow_step_size = self._get_hyper('slow_step_size', var_dtype)
         step_back = slow_var + slow_step_size * (var - slow_var)
-        sync_cond = tf.equal(local_step % sync_period, 0)
```

Comment:
> I would have assumed it'd fall back to CPU with dynamic kernels implemented.

I believe so. Have you ever tried `tf.math.floormod`?

Reply:
I've tried this one and it's not working.

Reply:
w/blob/d30ba8443710e94d6a1ab1c71a2c61c437e24790/tensorflow/core/kernels/cwise_op_gpu_floor_mod.cu.cc#L22

Reply:
Do you use tf.device to force all ops place on GPU? Can you give a minimal reproducing example? Thanks

Reply:
The outputs are the same as the results of this [CI](https://source.cloud.google.com/results/invocations/d11ec6bf-ede3-40cf-9c10-3112fe54ba12/targets). It fails the [fit_simple_linear_model](https://github.com/CyberZHG/addons/blob/b4cea8189724f88663637daa1df82d035d97cba4/tensorflow_addons/optimizers/lookahead_test.py#L110-L128) test.

Reply:
Thanks for the information, really useful! Could you file an issue for it and we might resolve it later?